test/common/mongo/proxy_test.cc

```diff
@@ -64,23 +67,122 @@ class MongoProxyFilterTest : public testing::Test {
 
     EXPECT_CALL(log_manager_, createAccessLog(_)).WillOnce(Return(file_));
     access_log_.reset(new AccessLog("test", log_manager_));
-    filter_.reset(new TestProxyFilter("test.", store_, runtime_, access_log_));
+  }
+
+  void initializeFilter() {
+    filter_.reset(new TestProxyFilter("test.", store_, runtime_, access_log_, fault_config_));
     filter_->initializeReadFilterCallbacks(read_filter_callbacks_);
     filter_->onNewConnection();
   }
 
+  void setupDelayFault(uint64_t duration_ms, uint32_t delay_percent, bool enable_fault) {
+    const std::string json_fault_config_template = R"EOF(
+    {{
+      "fixed_delay": {{
+        "percent": {},
+        "duration_ms": {}
+      }}
+    }}
+    )EOF";
+    const std::string json_config =
+        fmt::format(json_fault_config_template, delay_percent, duration_ms);
+    Json::ObjectSharedPtr config = Json::Factory::loadFromString(json_config);
+
+    fault_config_.reset(new FaultConfig(*config));
+
+    EXPECT_CALL(runtime_.snapshot_, featureEnabled(_, _)).Times(AnyNumber());
+    EXPECT_CALL(runtime_.snapshot_,
+                featureEnabled("mongo.fault.fixed_delay.percent", delay_percent))
+        .WillOnce(Return(enable_fault));
+
+    if (enable_fault) {
+      EXPECT_CALL(runtime_.snapshot_,
+                  getInteger("mongo.fault.fixed_delay.duration_ms", duration_ms))
+          .WillOnce(Return(duration_ms));
+    }
+  }
+
   Buffer::OwnedImpl fake_data_;
   NiceMock<TestStatStore> store_;
   NiceMock<Runtime::MockLoader> runtime_;
-  Event::MockDispatcher dispatcher_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
   std::shared_ptr<Filesystem::MockFile> file_{new NiceMock<Filesystem::MockFile>()};
   AccessLogSharedPtr access_log_;
+  FaultConfigSharedPtr fault_config_;
   std::unique_ptr<TestProxyFilter> filter_;
   NiceMock<Network::MockReadFilterCallbacks> read_filter_callbacks_;
   Envoy::AccessLog::MockAccessLogManager log_manager_;
 };
 
+TEST_F(MongoProxyFilterTest, DelayFaults) {
+  const uint64_t delay = 10;
+  setupDelayFault(delay, 50, true);
```

Comment:
All of your tests are using the same delay of 10, 50 just have on config that you reuse.

Reply:
fixed in the next iteration

Reply:
posted.